File: src/main/java/com/kishlaly/ta/analyze/testing/tp/TakeProfitDisabled.java
package com.kishlaly.ta.analyze.testing.tp;

import com.kishlaly.ta.model.SymbolDataJava;

public class TakeProfitDisabled extends TakeProfitStrategyJava {

    public TakeProfitDisabled() {
        super(null, false);
    }

    @Override
    public double calcualte(SymbolDataJava data, int signalIndex) {
        return -1;
    }

    @Override
    public boolean isEnabled() {
        return false;
    }

    @Override
    public String toString() {
        return "TP disabled";
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/testing/tp/TakeProfitDisabled.kt
package com.kishlaly.ta.analyze.testing.tp

import com.kishlaly.ta.model.SymbolData

class TakeProfitDisabled : TakeProfitStrategy() {

    override fun calculate(data: SymbolData, signalIndex: Int) = -1.0

    override fun toString() = "TP disabled"

}